src/v/cluster/partition_manager.cc

```diff
@@ -89,6 +89,19 @@ ss::future<consensus_ptr> partition_manager::manage(
     _ntp_table.emplace(log.config().ntp(), p);
     _raft_table.emplace(group, p);
     _manage_watchers.notify(p->ntp(), p);
+
+    /*
+     * part of the node draining infrastructure. when a node is in a drianing
+     * state new groups might be created since the controller will still be
+     * active as a follower. however, if draining is almost complete then new
+     * groups may not be noticed. marking as blocked should be done atomically
+     * with adding the partition to the ntp_table index above for proper
+     * synchronization with the drianing manager.
+     */
+    if (_block_new_leadership) {
+        p->block_new_leadership();
```

Comment:
should this have a timeout to unblock, say 30mins or 1hr or smth in case of a cluster split-brain after the draining started and can't make progress.